### PR TITLE
Fix checking of Heartbeat

### PR DESCRIPTION
**Pull request by VVeremjova**
Use local expectation of "Heartbeat" event instead of default one from ATF.
Use new functionality from ATF- start RPC with specific heartbeat

### CODE REVIEW

File: test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          EXPECT_HMICALL("UI.AddCommand"):Times(0)
+          EXPECT_HMICALL("VR.AddCommand"):Times(0)
+          EXPECT_RESPONSE(cid, { success = false, resultCode = "DISALLOWED" })
+
+          DelayedExp(10000)
+        end
+
+        --[[ Test ]]
+        function Test:TestStep_Get_HeartBeat_Time()
+          local function getHB()
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      end
+
+      function Test:Precondition_HMI_sends_OnAllowSDLFunctionality()
+        self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", {allowed = false, source = "GUI"})
+        self.mobileSession:ExpectNotification("OnPermissionsChange")
+      end
+
+      function Test:Precondition_Check_App_PreDataConsent()
+
+        self.mobileSession:ExpectEvent(eventHB, "Heartbeat"):Do(function(data)
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      :Do(function(_,data)
+          self.applications[config.application1.registerAppInterfaceParams.appName] = data.params.application.appID
+          self.HMIAppID = data.params.application.appID
+        end)
+      self.mobileSession:ExpectResponse(correlationId, { success = true, resultCode = "SUCCESS" })
+      self.mobileSession:ExpectNotification("OnHMIStatus", {hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+    end
+
+    function Test:Precondition_Activate_Consent_App()
+      self.mobileSession:ExpectEvent(eventHB, "Heartbeat"):Do(function(data)
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    eventHB = events.Event()
+    eventHB.matches = function(_, data)
+      return data.frameType == 0 and
+      (data.serviceType == 0) and
+      (data.frameInfo == 0) --HeartBeat
+    end
+
+    self.mobileSession:StartRPC(function (_, data)
+        self.mobileSession:ExpectEvent(eventHB, "Heartbeat")
+        :Do(function(data)
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    self.mobileSession.answerHeartbeatFromSDL = true
+
+    eventHB = events.Event()
+    eventHB.matches = function(_, data)
+      return data.frameType == 0 and
+      (data.serviceType == 0) and
+      (data.frameInfo == 0) --HeartBeat
+    end
+
+    self.mobileSession:StartRPC(function (_, data)
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/295_ATF_HP_Applying_Heart_Beat_Timeout_Ms_After_PTU.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua [hunk 6]
```diff
@@ -58,98 +58,104 @@ local function DelayedExp(time)
...
-          HBTime_max = diff
-          HBTime_min = diff
-        else
-          if HBTime_max < diff then HBTime_max = diff end
-          if HBTime_min > diff then HBTime_min = diff end
+    end
+
+    --[[ Test ]]
+    function Test:TestStep_Get_HeartBeat_Time()
+      local function getHB()
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua [hunk 7]
```diff
@@ -58,98 +58,104 @@ local function DelayedExp(time)
...
+
+    eventHB = events.Event()
+    eventHB.matches = function(_, data)
+      return data.frameType == 0 and
+      (data.serviceType == 0) and
+      (data.frameInfo == 0) --HeartBeat
+    end
+    self.mobileSession:StartRPC(function (_, data)
+        self.mobileSession:ExpectEvent(eventHB, "Heartbeat")
+        :Do(function(data)
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua [hunk 8]
```diff
@@ -58,98 +58,104 @@ local function DelayedExp(time)
...
+    self.mobileSession.sendHeartbeatToSDL = false
+    self.mobileSession.answerHeartbeatFromSDL = true
+
+    eventHB = events.Event()
+    eventHB.matches = function(_, data)
+      return data.frameType == 0 and
+      (data.serviceType == 0) and
+      (data.frameInfo == 0) --HeartBeat
+    end
+    self.mobileSession:StartRPC(function (_, data)
```
Comment (VVeremjova) [inline test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua]: @dboltovskyi See  89f30a6

File: test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua [hunk 9]
```diff
@@ -31,8 +30,9 @@ local testCasesForPolicyTable = require ('user_modules/shared_testcases/testCase
 local commonPreconditions = require ('user_modules/shared_testcases/commonPreconditions')
 
 --[[ Local Variables ]]
-local HBTime_max
-local HBTime_min
+local HBTime_max = 0
```
Comment (LuxoftAKutsan) [inline test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua]: @VVeremjova comments required

File: test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua [hunk 10]
```diff
@@ -66,36 +66,41 @@ end
...
-  event.matches = function(_, data)
-    return data.frameType == 0 and
-    (data.serviceType == 0) and
-    (data.frameInfo == 0) --HeartBeat
-  end
-
-  self.mobileSession:ExpectEvent(event, "Heartbeat")
+  self.mobileSession:ExpectEvent(eventHB, "Heartbeat")
   :ValidIf(function()
+      print("Heartbeat = "..timestamp().."ms")
```
Comment (LuxoftAKutsan) [inline test_scripts/Polices/Validation_of_PolicyTables/293_ATF_HP_Applying_Heart_Beat_Timeout_Ms_From_PT.lua]: @VVeremjova not clear output.

**Pull request merged**